test/unit_tests/autoqasm/test_api.py

```diff
@@ -1240,3 +1240,51 @@ def main():
 h __qubits__[2];
 h __qubits__[3];"""
     assert main.build().to_ir() == expected_ir
+
+
+def test_subroutine_call_with_kwargs():
+    """Test that subroutine calls works with keyword arguments"""
+
+    @aq.subroutine
+    def test(a: int, b: int) -> None:
+        aq.instructions.h(a)
+        aq.instructions.h(b)
+
+    @aq.main(num_qubits=2)
+    def main():
+        test(a=0, b=1)
+
+    expected = """OPENQASM 3.0;
+def test(int[32] a, int[32] b) {
+    h __qubits__[a];
+    h __qubits__[b];
+}
+qubit[2] __qubits__;
+test();"""
```

Comment:
I missed this on my first review. The correct output here would be `test(0, 1)`

Reply:
I don't see this when I run the code locally

Reply:
Reason for the error in the output: `args` were not updated after binding the `args` and `kwargs` to the function signature.
I have updated the code.